lib/internal/bootstrap_node.js

```diff
@@ -302,13 +302,13 @@
 
   function setupProcessFatal() {
 
-    process._fatalException = function(er) {
+    process._fatalException = function(er, fromPromise) {
       var caught;
 
       if (process.domain && process.domain._errorHandler)
         caught = process.domain._errorHandler(er) || caught;
 
-      if (!caught)
+      if (!caught && !fromPromise)
```

Comment:
@Fishrock123 We have definitely received bug reports of real-world V8 crashes because we tried to run JS code during GC, see b49b496a92118bf786176bc3c763a292b9bdff5d and its `Fixes:` tags